### DISCUSSION TITLE
Tasks marked overdue when due tomorrow

### DISCUSSION THREAD

**Discussion by aereaux** [TechbeeAT/jtxBoard#1439] [General]
I started using jtxBoard recently, and when it is late in the day and I have a task due tomorrow that task shows overdue.  Any idea what I could be doing wrong or if this is a bug?

**Comment by patrickunterwegs** [comment]
Hi @aereaux , how do you mean it's marked as overdue? Is it only red or does it say 'overdue'? Can you also tell me in which timezone you are?

**Reply by aereaux** [reply]
Thanks for the response!  I'm in PDT (UTC-7).  It is red and says overdue, same as it appears when it is actually overdue

**Reply by patrickunterwegs** [reply]
Thanks, I'll check. Those things are really hard to test, but I've seen in the code already that for due dates without time there could be an issue. Just to be sure, you're talking about due dates without time or does this also happen when you have a time?

**Reply by aereaux** [reply]
Thanks, let me know if there's anything you'd like me to test on my end.  Yes, this is with just a due date set, without a time.

**Reply by patrickunterwegs** [reply]
Hi @aereaux , indeed it would be great if you could check if the issue is solved in the latest beta. It is published on 
Github: https://github.com/TechbeeAT/jtxBoard/releases/tag/v2.7.9-beta01
and probably tomorrow it will also be approved on the Google Play beta channel.

**Reply by aereaux** [reply]
OK, I installed from f-droid previously so I'll need to uninstall and reinstall.  Doing that will resync all my tasks from davx5, right?

**Reply by patrickunterwegs** [reply]
Yep, just make sure to sync before you uninstall. After reinstalling you need to grant the sync permission again within DAVx5, but you'll get a notification anyway.

**Reply by aereaux** [reply]
I don't think this helped at all.  The tasks due today and tomorrow are no longer marked in red, but the ones due tomorrow are still marked "overdue"

**Reply by patrickunterwegs** [reply]
Thanks, I'll revert the change. 

Could you send me a screenshot of such a problematic every (list and details)?

**Reply by aereaux** [reply]
If it serves as a clue, shortly before midnight UTC it says "due shortly".  Here's a couple screenshots, let me know if this isn't exactly what you were expecting:
![zb2rhm5dSftLvFURAY4uUZdbNNUpX5mZQoagBs39MyhQAYmsA](https://github.com/TechbeeAT/jtxBoard/assets/8238401/eda649af-99b5-4764-a846-63364af53f7b)
![IMG_20240530_171833](https://github.com/TechbeeAT/jtxBoard/assets/8238401/8dbd179f-5fa6-4680-b465-912b6ad92303)
![IMG_20240530_171520](https://github.com/TechbeeAT/jtxBoard/assets/8238401/57e2ed00-b258-4e0d-9f30-257901afda99)

**Reply by patrickunterwegs** [reply]
Thanks @aereaux ! It's still not very clear to me what causes the issue. It would also be interesting to see the actual time in the screenshots. I have checked again and made another small change, that possibly could be the cause of this issue. May I ask you to try/observe again with 2.7.9-beta02? I've just uploaded it. Thank you!

**Reply by aereaux** [reply]
I installed the -beta04 that was the most recent when I checked.  Does that have the fix you mentioned?  In any case with that installed they now say "due today" when it is still the end of the previous day.

And I took the screenshots pretty soon before posting them.  I think it was a little after 5pm local time, which is a bit after midnight UTC the next day.

**Reply by patrickunterwegs** [reply]
Thanks again for the feedback @aereaux !
Once more, may I ask you to check [v2.7.9-beta07](https://github.com/TechbeeAT/jtxBoard/releases/tag/v2.7.9-beta07) ?

So my assumption is that the calculation of the information if it's overdue is calculated (wrongly) based on the UTC time. In this case - as you are in GMT-7, I assume that the app shows correctly "Due tomorrow" until 5pm, can you confirm that? 
Can you tell me how you have set up your time and timezone? Did you set it up manually or just take it from the system?

**Reply by aereaux** [reply]
No problem!  I installed beta07 yesterday when I saw it, and it seems like it still doesn't work.  Now I think the items with due dates of tomorrow say due today after 5pm local time.  I'll reconfirm after 5pm here though.

The app does indeed say "Due tomorrow" right now, and I think it switches to "due shortly" closer to 5pm, but I'll reconfirm that as well.

I didn't do any explicit time or timezone configuration, so it should just be taking it from the system.

**Reply by patrickunterwegs** [reply]
Thanks for the feedback! 
Sorry that this is such a hard thing to figure out! But I have another idea now. Could you confirm that the functionality works as expected when there is a time? So it's only an issue for entries with due dates without time, right?

**Reply by aereaux** [reply]
I'm happy to provide any debugging needed.  Would it matter what the current time is?  Currently it looks like whatever time I put it correctly reports whether the item is overdue.

**Reply by patrickunterwegs** [reply]
Thank you! Yes, your local time does matter. So it would be great if you could observe this also after 5pm of your local time! =)
In the meantime I'll upload a new version with another change that now might be closer to what's needed.

**Reply by patrickunterwegs** [reply]
> Yes, your local time does matter. 
Sorry, to get this right: Your local 5 pm is the GMT midnight (and switch of the day), so it would be great if you could check with an entry due before 5 pm and after 5 pm!

**Reply by aereaux** [reply]
Yeah, the ones with times on them show correctly "due in X hours" whether they are before or after midnight GMT.  Interestingly when I put 4pm local time it displays sorted before the events due tomorrow, but when I put 6pm local time it displays them after the events tomorrow.  Hopefully that helps, and I can check for events when the current time is after 5pm as well.

**Reply by aereaux** [reply]
I'm not sure if there are any other side effects, but `beta08` does seem to have fixed the issue I was having.  Thanks for the work on this!

**Reply by patrickunterwegs** [reply]
That's good news, thank you :-) Side effects should be minimal, because I added a condition to specifically treat the check if overdue differently when there's no time. Thanks again for your support =)
About your other comment about the sort order: This was already addressed in #995, back then I couldn't find a good solution, but now I have something in mind how I can solve the problem more easily. I will tackle that soon.
